Added localization

hwckent opened the pull request:
English translation

Chinese translation

ManlyMarco commented:
I'm not sure about this one to be honest, this adds a lot of extra complexity and new things to maintain.

The currently recommended way of translating the plugin is by using AutoTranslator, as is done for Japanese plugins like [KK_ExpressionControl](https://github.com/IllusionMods/KoikatsuTranslation/blob/master/translation/en/Text/Plugins/KK_ExpressionControl/Translation.txt). It doesn't require any code changes in the plugin and allows adding new languages very easily.

hwckent commented:
Introduced Newtonsoft.Json: This is a widely used .NET library for serializing and deserializing JSON, which is very suitable for loading external language files.
Externalized language data: The text of the plugin interface is localized by loading translations from the languages.json file. This means that you can easily add or modify translations without recompiling the plugin.
T(string key) translation function: This is a concise and efficient implementation that finds the translation of the corresponding language by the key name. If the translation is not found, it will fall back to the original key name, which is helpful for debugging and identifying missing translations.
Language selector (DrawLanguageSelector): Added a language switch button in the GUI, so that users can easily choose to display Chinese or English, which greatly improves the user experience.
Globally apply translations: You replace all hard-coded English strings with T("key") to ensure that most of the text in the plugin interface can be translated.
Error handling: Added exception handling when the languages.json file fails to load, and logs, which improves the robustness of the plugin.
